lib/auth/auth.go

```diff
@@ -798,6 +815,22 @@ func (a *AuthServer) DeleteRole(name string) error {
 	return a.Access.DeleteRole(name)
 }
 
+// recordEvent records the provided event using auth server recorder if it's set
+func (a *AuthServer) recordEvent(event reporting.Event) {
+	if a.eventRecorder == nil {
+		return
+	}
+	a.eventRecorder.Record(event)
+}
+
+// recordUserEvent anonymizes the user information and records an event
+func (a *AuthServer) recordUserEvent(email string) {
+	h := hmac.New(sha256.New, []byte(a.clusterID))
```

Comment:
can you add a note why are we adding clusterID and another note to folks reading code that OSS does not  have reporting on.